ts/client/scripts/risk.ts

```diff
@@ -0,0 +1,469 @@
+      toUiDecimalsForQuote(liabsInUsdc).toFixed(2),
+      (pi1.priceImpactPct * 100).toFixed(2),
+      (toUiDecimals(assets, bank.mintDecimals) * bank.uiPrice).toFixed(2),
+      (pi2.priceImpactPct * 100).toFixed(2),
+      parseFloat(
+        res['lastXTopTokens'].filter(
+          (entry) => entry.mint === bank.mint.toBase58(),
+        )[0].amount,
+      ).toFixed(2),
+    ]);
+  }
+  stream.end();
+  console.log(title);
+  table.printTable();
+  console.log();
+}
+
+async function computePerpPositionsToBeLiquidated(
+  group: Group,
+  mangoAccounts: MangoAccount[],
+  healthThresh: number,
+  title: string,
+  csvSuffix: string,
+): Promise<void> {
+  const mangoAccountsWithHealth = mangoAccounts
+    .map((a: MangoAccount) => {
+      return {
+        account: a,
+        health: a.getHealth(group, HealthType.maint),
+        healthRatio: a.getHealthRatioUi(group, HealthType.maint),
+        liabs: toUiDecimalsForQuote(a.getLiabsValue(group, HealthType.maint)),
+      };
+    })
+    .filter((a) => a.healthRatio < healthThresh);
+
+  const table = new Table({
+    columns: [
+      { name: 'Market', alignment: 'right' },
+      { name: 'Price', alignment: 'right' },
+      { name: 'Future Price', alignment: 'right' },
+      { name: 'Notional Position', alignment: 'right' },
+    ],
+  });
+  const fileName = `/tmp/${
+    new Date().toISOString().split('T')[0]
+  }-${csvSuffix}-perp_market.csv`;
+  const csvFile = fs.createWriteStream(fileName);
+  const stream = format({ headers: true });
+  stream.pipe(csvFile);
+  stream.write(['Market', 'Price', 'Future Price', 'Notional Position']);
+
+  for (const pm of Array.from(group.perpMarketsMapByMarketIndex.values())) {
+    // todo: rather simply much position you would liquidate unweighted_health_per_lot
+    // to offset negative health so that health comes to 0
+    const baseLots = mangoAccountsWithHealth
+      .filter((a) => a.account.getPerpPosition(pm.perpMarketIndex))
+      .reduce((sum, a) => {
+        const baseLots = a.account.getPerpPosition(
+          pm.perpMarketIndex,
+        )!.basePositionLots;
+        const unweightedHealthPerLot = baseLots.gt(new BN(0))
+          ? I80F48.fromNumber(-1)
+              .mul(pm.price)
+              .mul(I80F48.fromU64(pm.baseLotSize))
+              .mul(pm.initBaseAssetWeight)
+              .add(
+                I80F48.fromU64(pm.baseLotSize)
+                  .mul(pm.price)
+                  .mul(
+                    ONE_I80F48() // quoteInitAssetWeight
+                      .mul(ONE_I80F48().sub(pm.baseLiquidationFee)),
+                  ),
+              )
+          : pm.price
+              .mul(I80F48.fromU64(pm.baseLotSize))
+              .mul(pm.initBaseLiabWeight)
+              .sub(
+                I80F48.fromU64(pm.baseLotSize)
+                  .mul(pm.price)
+                  .mul(ONE_I80F48()) // quoteInitLiabWeight
+                  .mul(ONE_I80F48().add(pm.baseLiquidationFee)),
+              );
```

Comment:
these two look right! Could simplify to
`price * baselotsize * ((1+fee) - baseassetweight)` and `price * baselotsize * (baseliabweight - (1-fee))`